addons/mrp/migrations/15.0.2.1/pre-migration.py

```diff
@@ -0,0 +1,18 @@
+def migrate(cr, version):
+    cr.execute(
+        """
+        WITH production_info AS (
+            SELECT mp.id,
+                   MIN(wo.date_start) AS date_start
+              FROM mrp_production mp
+              JOIN mrp_workorder wo
+                ON wo.production_id = mp.id
+             WHERE mp.state != 'draft'
+               AND mp.date_start IS NULL
+             GROUP BY mp.id
+        ) UPDATE mrp_production mp
+             SET date_start = info.date_start
+            FROM production_info info
+           WHERE mp.id = info.id
```

Comment:
For readability
```suggestion
        )
        UPDATE mrp_production mp
            SET date_start = info.date_start
          FROM production_info info
          WHERE mp.id = info.id
```

Reply:
@davidtranhp Styles suggested in the comments above https://github.com/odoo/odoo/pull/128043#discussion_r1259418764

Reply:
Which one?
When define CTE, the style should be
```sql
WITH cte1 AS (

),
cte2 AS (

)
SELECT ......
```
It is easier to read than
```sql
WITH cte1 AS (

),
cte2 AS (

) SELECT ......
```

Reply:
Yes sir. I fixed it